nshlib/nsh_printf.c

```diff
@@ -84,7 +87,34 @@ int cmd_printf(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
             {
               case 'x':
                 fmt++;
-                nsh_output(vtbl, "%c", strtol(fmt, NULL, 16));
+                value = strtol(fmt, NULL, 16);
+                len = strnlen(fmt, 10);
+
+                /* Is it a Big Endian MCU ? */
+
+                if (BYTE_ORDER == BIG_ENDIAN)
+                  {
+                    value = __swap_uint32(value);
+                  }
+
+                if (len == 8)
+                  {
+                    nsh_output(vtbl, "%c%c%c%c", value & 0xff,
+                                                 (value & 0xff00) >> 8,
+                                                 (value & 0xff0000) >> 16,
+                                                 (value & 0xff000000) >> 24);
+                  }
+                else
+                  if (len == 4)
```

Comment:
Why split "else if" into two lines?

Reply:
I use to do this way, but I could keep it in a single line.